spec/features/post_show_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe 'Post show feature' do
+  user = User.includes(:posts).first
+  post = user.posts.first
+
+  before(:each) do
+    visit(user_post_path(user.id, post.id))
+  end
+
+  scenario 'Confirm that we see the post title' do
+    expect(page).to have_content(post.Title.to_s)
+  end
+
+  scenario 'Confirm we can see who wrote the post' do
+    expect(page).to have_content(user.Name.to_s)
+  end
+
+  scenario 'Confirm that we can see how many comments the post has' do
+    expect(page).to have_content("Comments: #{post.comments.count}")
+  end
+
+  scenario 'Confirm that we see how many likes the post has' do
+    expect(page).to have_content("Likes: #{post.likes.count}")
+  end
+
+  scenario 'Confirm to see that we have the post body' do
+    expect(page).to have_content(post.Text.to_s)
+  end
+
+  scenario 'Confirm that we can see the username of each commentor' do
+    post.comments.includes(:user).each do |comment|
+      expect(page).to have_content("#{comment.user.Name} : ")
+    end
+  end
+
+  scenario 'Confirm that we can see the comment body for each user' do
+    post.comments.includes(:user).each do |comment|
+      expect(page).to have_content("#{comment.user.Name} : #{comment.Text}")
+    end
+  end
+end
```

Comment:
- [ ] Weldon guys!, you've done a good job so far, however, this `spec/features/post_show_spec.rb` tests failed because you didn't create a `user` and `posts` before making use of them. What we can do to fix this is to create a user, then create posts for that user and the test will pass. See what I did and the error was gone.

```suggestion
require 'rails_helper'

RSpec.describe 'Post show feature' do
  before(:each) do
   @user = FactoryBot.create(:user)
    m = 5
    while m > 0
      FactoryBot.create(:post, user_id: @user.id)
      m -= 1
    end
    @post = @user.posts.first
    visit(user_post_path(@user.id, @post.id))
  end

  scenario 'Confirm that we see the post title' do
    expect(page).to have_content(@post.Title.to_s)
  end

  scenario 'Confirm we can see who wrote the post' do
    expect(page).to have_content(@user.Name.to_s)
  end

  scenario 'Confirm that we can see how many comments the post has' do
    expect(page).to have_content("Comments: #{@post.comments.count}")
  end

  scenario 'Confirm that we see how many likes the post has' do
    expect(page).to have_content("Likes: #{@post.likes.count}")
  end

  scenario 'Confirm to see that we have the post body' do
    expect(page).to have_content(@post.Text.to_s)
  end

  scenario 'Confirm that we can see the username of each commentor' do
    @post.comments.includes(:user).each do |comment|
      expect(page).to have_content("#{comment.user.Name} : ")
    end
  end

  scenario 'Confirm that we can see the comment body for each user' do
    @post.comments.includes(:user).each do |comment|
      expect(page).to have_content("#{comment.user.Name} : #{comment.Text}")
    end
  end
end
```

Reply:
Thank you so much for pointing this out @julie-ify, It was working on local machine so we relaxed. Changes will be implemented